### PR TITLE
Adjust to netlib changes

### PR DESCRIPTION
**Pull request by mhils**
refs https://github.com/mitmproxy/netlib/pull/92

### CODE REVIEW

File: libpathod/pathoc.py [hunk 1]
```diff
@@ -288,9 +289,9 @@ def connect(self, connect_to=None, showssl=False, fp=sys.stdout):
         self.sslinfo = None
         if self.ssl:
             try:
-                alpn_protos = [HTTP1Protocol.ALPN_PROTO_HTTP1]
+                alpn_protos = [ALPN_PROTO_HTTP1]
```
Comment (Kriechi) [inline libpathod/pathoc.py]: why not `http.ALPN_PROTO_HTTP1` as two lines below?

File: libpathod/protocols/http.py [hunk 2]
```diff
@@ -70,4 +68,8 @@ def handle_http_connect(self, connect, lg):
         return self.pathod_handler.handle_http_request, None
 
     def read_request(self, lg=None):
-        return self.wire_protocol.read_request(allow_empty=True)
+        try:
+            return http1.read_request(self.pathod_handler.rfile)
+        except HttpReadDisconnect:
+            # TODO: This is
+            return Request("", b"", b"", b"", b"", b"", b"", None, b"")
```
Comment (Kriechi) [inline libpathod/protocols/http.py]: That's why I added `EmptyRequest` - any reason why not to keep it?

Comment (mhils) [inline libpathod/protocols/http.py]: I did not really understand the purpose of this in netlib. What should an `EmptyRequest` express? 
The situation here is ugly indeed, we should improve this.

File: libpathod/pathoc.py (hunk 1 above)
Comment (mhils) [inline libpathod/pathoc.py]: :+1:

File: libpathod/protocols/http.py (hunk 2 above)
Comment (mhils) [inline libpathod/protocols/http.py]: Fixed - we now let the HttpReadDisconnect error  go through and act here accordingly.

**Pull request merged**